expression/builtin_json.go

```diff
@@ -733,8 +735,34 @@ type jsonQuoteFunctionClass struct {
 	baseFunctionClass
 }
 
+type builtinJSONQuoteSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJSONQuoteSig) Clone() builtinFunc {
+	newSig := &builtinJSONQuoteSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
 func (c *jsonQuoteFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_QUOTE")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETJson)
+	args[0].GetType().Flag &= ^mysql.ParseToJSONFlag
+	sig := &builtinJSONQuoteSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_JsonQuoteSig)
+	return sig, nil
+}
+
+func (b *builtinJSONQuoteSig) evalString(row chunk.Row) (res string, isNull bool, err error) {
+	var j json.BinaryJSON
+	j, isNull, err = b.args[0].EvalJSON(b.ctx, row)
+	if isNull || err != nil {
+		return "", isNull, errors.Trace(err)
+	}
+	return j.Quote(), false, nil
```

Comment:
We’d better do a deep copy for the result of function ‘Quote’.

Reply:
@zz-jason Why should we make a deep copy for a string?

Reply:
Because in `BinaryJson.Quote()`, the returned string is only a shallow copy of the binary json, and this function also returns the shallow copied string.

The upper layer caller may hold this string reference. But if the content of the deep layer binary json is changed, the content of that string reference is changed.